classes/i18n/Locale.php

```diff
@@ -353,6 +353,21 @@ public function getScripts(?string $locale = null): Scripts
         return $this->_getLocaleCache(__METHOD__, $locale, fn () => $this->_getIsoCodes($locale)->getScripts());
     }
 
+    /**
+     * Get the locales counts by only locale code
+     */
+    public function getLocaleCodesCount(array $allLocales = []): array
```

Comment:
This is a general coding style preference rather than something specific to this function, but I prefer to use an explicit `null` for special cases rather than an empty array for an array parameter. That way if the `$allLocales` parameter comes from an upstream array operation of some kind that legitimately returns a `[]`, the function will behave as expected.

Reply:
This is kind of a confusing function -- it's not clear by looking at its description what it does or how it might be used. Both of the use cases for it carry some duplicated code, so I suggest making this function do a little more -- e.g. get a list of language codes by display name. That'll both clear up this function and reduce the duplicated code in both calling cases.

Reply:
Done .